src/common/libflux/reactor.c

```diff
@@ -563,6 +563,26 @@ flux_buffer_t *flux_buffer_write_watcher_get_buffer (flux_watcher_t *w)
     return NULL;
 }
 
+void flux_buffer_write_watcher_close (flux_watcher_t *w)
+{
+    if (w) {
+        struct ev_buffer_write *evw = w->data;
+        evw->eof = true;
```

Comment:
Should we add  ```if (!evw->eof && !evw->closed)``` so users can't close multiple times?

Reply:
I thought it should be idempotent. However, you are right might as well check for evw->closed otherwise eof is reset and the close callback could be called again. 

Should the function therefore have a return value and return EINVAL if `evw->closed` and EINPROGRESS if `evw->eof && !evw->closed`?

Reply:
Yeah, that's what I was thinking, that the close callback could be called multiple times.  This function returns void, so should the errno matter?